src/main/java/org/tarantool/TarantoolClientImpl.java

```diff
@@ -165,6 +177,15 @@ protected void connect(final SocketChannel channel) throws Exception {
 
     protected void startThreads(String threadName) throws InterruptedException {
         final CountDownLatch init = new CountDownLatch(2);
+
+        if (reader != null) {
+            reader.join(config.initTimeoutMillis / 2);
+        }
+        if (writer != null) {
+            writer.join(config.initTimeoutMillis / 2);
+        }
+        state.release(StateHelper.RECONNECT);
```

Comment:
Hm, it did not released before this patch? To be honest, don't understood this change :(

Reply:
see a comment left above.
This action replaces an old `state.compareAndSet(RECONNECT, 0)` invoked at the beginning of the connector thread.

Reply:
Now undestood. Thanks.